brendel-group/mission_db
# Restapi Api Key Authorization

This PR adds authorization for the REST API.\
I used [this](https://florimondmanca.github.io/djangorestframework-api-key/) module for this.\
Now every request to the REST API has to have an Authorization header in the format
```
Authorization: Api-Key <API_KEY>
```
This can be disabled by setting `DEBUG=True` in the `backend/backend/.env` file.\
Unfortunately the browsable API won't work with this.

API KEYs can be added or removed via the CLI.

The key itself is not stored in the database, just the hash value.

I made changes and added the documentation for the [CLI](https://github.com/brendel-group/mission_db/blob/feature/restapi-api-key/docs/cli/README.md), the [frontend](https://github.com/brendel-group/mission_db/blob/feature/restapi-api-key/docs/frontend/README.md) and the [restapi](https://github.com/brendel-group/mission_db/blob/feature/restapi-api-key/docs/restapi/README.md).

I changed the frontend to use an API KEY if it is set in a new `.env` file in the `frontend/` folder like this:
```bash
VITE_BACKEND_API_KEY = '<API_KEY>'
```

This resolves #136 

## How to test:
To not use any API KEYs start the backend with `DEBUG=True` and confirm that everything works like before.

To use API KEYs set `DEBUG=False` in the `.env` file for the backend, add an API KEY with the CLI, add it to the `.env` file for the frontend and test the frontend.

To manually test the REST API you can use curl like this:
```bash
curl -H "Authorization: Api-Key <API_KEY>" localhost:8000/restapi/missions/
```
to make other requests than GET you can use for example the option `-X DELETE` to make a DELETE request.
